libsentrykube/datadog.py

```diff
@@ -42,13 +47,22 @@ def check_monitors(
 
 
 def check_monitor(
-    monitor_id: int, dd_app_key: str | None = None, failure_states: Sequence[str] | None = None
+    monitor_id: int,
+    dd_app_key: str | None = None,
+    failure_states: Sequence[str] | None = None,
 ) -> bool:
     if dd_app_key is None:
-        dd_app_key = os.getenv("DATADOG_APP_KEY") or os.getenv("DD_APP_KEY") or None
+        dd_app_key = os.getenv("DATADOG_APP_KEY") or os.getenv("DD_APP_KEY")
 
     if dd_app_key is None:
-        raise MissingDataDogAppKeyException("DATADOG_APP_KEY must be set to check monitors.")
+        raise MissingDataDogAppKeyException(
+            "DATADOG_APP_KEY must be set to check monitors."
+        )
+
+    if DATADOG_API_KEY is None:
+        raise MissingDataDogApiKeyException(
+            "DATADOG_API_KEY must be set to check monitors."
+        )
```

Comment:
Need this check because [headers](https://github.com/getsentry/sentry-infra-tools/pull/42/files#diff-a91772c98fa3fc875eebd1169267d21e659e4c80c24b11c745f5793b7cb206d6R72) needs to be a dict of str,str. But DATADOG_API_KEY can be none right now